tests/core/test_memory.py

```diff
@@ -247,3 +262,25 @@ def forward(self, *args, **kwargs):
     model.example_input_array = example_input
     summary = model.summarize(mode=mode)
     assert summary.in_sizes == [expected_size]
+
+@pytest.mark.parametrize(['mode'], [
+    pytest.param(ModelSummary.MODE_FULL),
+    pytest.param(ModelSummary.MODE_TOP),
+])
+def test_model_size(mode):
+    """ Test that model size is calculated correctly. """
+    model = PreCalculatedModel()
+    summary = model.summarize(mode=mode)
+    pre_calculated_model_size = torch.tensor(model.pre_calculated_model_size)
+    model_size = torch.tensor(summary.model_size)
+    assert torch.isclose(model_size, pre_calculated_model_size, atol=1e-4)
```

Comment:
perhaps it could be easier to just make a 1 megabyte model and assert the exact value here, instead of the approximation :)

Reply:
@awaelchl nice suggestion :) thanks

Reply:
done